diff -urN a/bus/org.freedesktop.dbus-session.plist.in b/bus/org.freedesktop.dbus-session.plist.in
--- a/bus/org.freedesktop.dbus-session.plist.in	2022-02-23 22:39:11.000000000 +0800
+++ b/bus/org.freedesktop.dbus-session.plist.in	2022-06-21 13:04:41.286543144 +0800
@@ -8,12 +8,6 @@
 	<key>ServiceIPC</key>
 	<true/>
 
-	<!-- Please uncomment on 10.4; OnDemand doesn't work properly there. -->
-	<!--
-	<key>OnDemand</key>
-	<false />
-	-->
-
 	<key>ProgramArguments</key>
 	<array>
 		<string>@DBUS_DAEMONDIR@/dbus-daemon</string>
@@ -29,5 +23,7 @@
 			<string>DBUS_LAUNCHD_SESSION_BUS_SOCKET</string>
 		</dict>
 	</dict>
+	<key>UserName</key>
+	<string>messagebus</string>
 </dict>
 </plist>
diff -urN a/bus/system.conf.in b/bus/system.conf.in
--- a/bus/system.conf.in	2022-02-23 22:39:11.000000000 +0800
+++ b/bus/system.conf.in	2022-06-21 13:01:30.329757736 +0800
@@ -26,9 +26,6 @@
   <!-- This is a setuid helper that is used to launch system services -->
   <servicehelper>@DBUS_LIBEXECDIR@/dbus-daemon-launch-helper</servicehelper>
 
-  <!-- Write a pid file -->
-  <pidfile>@DBUS_SYSTEM_PID_FILE@</pidfile>
-
   <!-- Enable logging to syslog -->
   <syslog/>
 
