diff -urN a/dlls/winecoreaudio.drv/coreaudio.c d/dlls/winecoreaudio.drv/coreaudio.c
--- a/dlls/winecoreaudio.drv/coreaudio.c	2022-06-18 03:52:59.000000000 +0800
+++ d/dlls/winecoreaudio.drv/coreaudio.c	2022-06-24 02:29:52.000000000 +0800
@@ -444,7 +444,7 @@
 
     memset(&desc, 0, sizeof(desc));
     desc.componentType = kAudioUnitType_Output;
-    desc.componentSubType = kAudioUnitSubType_HALOutput;
+    desc.componentSubType = 'ahal';
     desc.componentManufacturer = kAudioUnitManufacturer_Apple;
 
     if(!(comp = AudioComponentFindNext(NULL, &desc))){
diff -urN a/dlls/winecoreaudio.drv/coremidi.c d/dlls/winecoreaudio.drv/coremidi.c
--- a/dlls/winecoreaudio.drv/coremidi.c	2022-06-18 03:52:59.000000000 +0800
+++ d/dlls/winecoreaudio.drv/coremidi.c	2022-06-24 03:05:48.000000000 +0800
@@ -458,7 +458,7 @@
 
     /* create synth node */
     desc.componentType = kAudioUnitType_MusicDevice;
-    desc.componentSubType = kAudioUnitSubType_DLSSynth;
+    desc.componentSubType = 'dls ';
 
     sc = AUGraphAddNode(*graph, &desc, &synth_node);
     if (sc != noErr)
@@ -469,7 +469,7 @@
 
     /* create out node */
     desc.componentType = kAudioUnitType_Output;
-    desc.componentSubType = kAudioUnitSubType_DefaultOutput;
+    desc.componentSubType = (1L << 2);
 
     sc = AUGraphAddNode(*graph, &desc, &out_node);
     if (sc != noErr)
@@ -831,7 +831,7 @@
         float left;
 
         if (!once++) FIXME("independent left/right volume not implemented\n");
-        AudioUnitGetParameter(dests[dev_id].synth, kHALOutputParam_Volume, kAudioUnitParameterFlag_Output, 0, &left);
+        AudioUnitGetParameter(dests[dev_id].synth, kHALOutputParam_Volume, (1L << 2), 0, &left);
         *volume = (WORD)(left * 0xffff) + ((WORD)(left * 0xffff) << 16);
         return MMSYSERR_NOERROR;
     }
@@ -857,7 +857,7 @@
         left  = LOWORD(volume) / 65535.0f;
         right = HIWORD(volume) / 65535.0f;
 
-        AudioUnitSetParameter(dests[dev_id].synth, kHALOutputParam_Volume, kAudioUnitParameterFlag_Output, 0, left, 0);
+        AudioUnitSetParameter(dests[dev_id].synth, kHALOutputParam_Volume, (1L << 2), 0, left, 0);
         return MMSYSERR_NOERROR;
     }
 
